contracts/strategies/lift/LiftStrategy.sol

```diff
@@ -0,0 +1,272 @@
+*   6. YFIGovernance https://etherscan.io/address/0xBa37B002AbaFDd8E89a1995dA52740bbC013D992#code
+*
+*
+*
+*   Respecting the current system design of choosing the best strategy under the vault, and also rewarding/funding
+*   the public key that invokes the switch of strategies, this smart contract should be deployed twice and linked
+*   to the same vault. When the governance want to rotate the crop, they would set the reward source on the strategy
+*   that is not active, then set that apy higher and this one lower.
+*
+*   Consequently, in the smart contract we restrict that we can only set a new reward source when it is not active.
+*
+*/
+
+contract LiftStrategy is StrategyBaseClaimable {
+
+  using SafeMath for uint256;
+  using SafeERC20 for IERC20;
+
+  address public uniLPComponentToken0;
+  address public uniLPComponentToken1;
+  address public ctrl;
+
+  bool public pausedInvesting = false; // When this flag is true, the strategy will not be able to invest. But users should be able to withdraw.
+
+  ISharePool public rewardPool;
+  IBoardRoom public boardRoom;
+
+  uint256[2][] stakes;
+
+  event ProfitsNotCollected();
+
+  // This is only used in `investAllUnderlying()`
+  // The user can still freely withdraw from the strategy
+  modifier onlyNotPausedInvesting() {
+    require(!pausedInvesting, "Action blocked as the strategy is in emergency state");
+    _;
+  }
+
+  constructor(
+    address _storage,
+    address _underlying,
+    address _vault,
+    address _rewardPool,
+    address _rewardToken,
+    address _uniswapRouterV2
+  )
+  StrategyBaseClaimable(_storage, _underlying, _vault, _rewardToken, _rewardToken, _uniswapRouterV2)
+  public {
+    uniLPComponentToken0 = IUniswapV2Pair(underlying).token0();
+    uniLPComponentToken1 = IUniswapV2Pair(underlying).token1();
+    rewardPool = ISharePool(_rewardPool);
+    boardRoom = IBoardRoom(rewardPool.boardroom());
+    ctrl = boardRoom.control();
+  }
+
+  function depositArbCheck() public view returns(bool) {
+    return true;
+  }
+
+  /*
+  *   In case there are some issues discovered about the pool or underlying asset
+  *   Governance can exit the pool properly
+  *   The function is only used for emergency to exit the pool
+  */
+  function emergencyExit() public onlyGovernance {
+    rewardPool.exit();
+    pausedInvesting = true;
+  }
+
+  /*
+  *   Resumes the ability to invest into the underlying reward pools
+  */
+
+  function continueInvesting() public onlyGovernance {
+    pausedInvesting = false;
+  }
+
+  /*
+  *   Stakes everything the strategy holds into the reward pool
+  */
+  function investAllUnderlying() internal onlyNotPausedInvesting {
+    // this check is needed, because most of the SNX reward pools will revert if
+    // you try to stake(0).
+    if(IERC20(underlying).balanceOf(address(this)) > 0) {
+      IERC20(underlying).approve(address(rewardPool), IERC20(underlying).balanceOf(address(this)));
```

Comment:
Can we try `safeApprove` for consistency?

Reply:
Will update

Reply:
Fixed